src/net_processing.cpp

```diff
@@ -614,7 +614,7 @@ class PeerManagerImpl final : public PeerManager
      * and in best equivalent proof of work) than the best header chain we know
      * about and we fully-validated them at some point.
      */
-    bool BlockRequestAllowed(const CBlockIndex* pindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+    bool BlockRequestAllowed(const CBlockIndex* pindex, bool check_valid_scripts = true) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
I'm not sure this is a good var name. This specific PR has nothing to do with valid scripts for example.